src/EventStore.Core/Services/Archiver/ChunkPrioritizer.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Event Store Ltd and/or licensed to Event Store Ltd under one or more agreements.
+// Event Store Ltd licenses this file to you under the Event Store License v2 (see LICENSE.md).
+
+using System.Collections.Generic;
+
+namespace EventStore.Core.Services.Archiver;
+
+public class ChunkPrioritizer : IComparer<Commands.ArchiveChunk> {
+	public int Compare(Commands.ArchiveChunk x, Commands.ArchiveChunk y) {
+		int cmp = x!.ChunkStartNumber.CompareTo(y!.ChunkStartNumber);
```

Comment:
are the `!` needed here

Reply:
yes, because that's how IComparer is defined:
```
  public interface IComparer<in T>
  {
    int Compare(T? x, T? y);
  }
```

[strictly speaking](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.icomparer-1.compare?view=net-8.0#remarks) we should implement the case where x and y can be null but in this case it should never happen

Reply:
i think thats with nullable enabled though, x and y are not nullable in ChunkPrioritizer.cs

> public int Compare(Commands.ArchiveChunk x, Commands.ArchiveChunk y)

probably the best thing would be to `#nullable enable` the file (this would make sense as routine for new files)

Reply:
fixed :+1: